src/ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,615 @@
+"""Tool to generate IRFs"""
+
+import operator
+from functools import partial
+
+import astropy.units as u
+import numpy as np
+from astropy.io import fits
+from astropy.table import QTable, vstack
+from pyirf.cuts import evaluate_binned_cut
+from pyirf.io import create_rad_max_hdu
+
+from ..core import Provenance, Tool, ToolConfigurationError, traits
+from ..core.traits import AstroQuantity, Bool, Integer, classes_with_traits, flag
+from ..irf import (
+    EventPreProcessor,
+    EventsLoader,
+    OptimizationResultStore,
+    Spectra,
+    check_bins_in_range,
+)
+from ..irf.benchmarks import (
+    AngularResolutionMakerBase,
+    EnergyBiasResolutionMakerBase,
+    SensitivityMakerBase,
+)
+from ..irf.irfs import (
+    BackgroundRateMakerBase,
+    EffectiveAreaMakerBase,
+    EnergyMigrationMakerBase,
+    PsfMakerBase,
+)
+
+
+class IrfTool(Tool):
+    name = "ctapipe-make-irf"
+    description = "Tool to create IRF files in GAD format"
+
+    do_background = Bool(
+        True,
+        help="Compute background rate IRF using supplied files",
```

Comment:
we should be clear with naming here and follow the data model terms.

 *The IRF* is the whole thing, i.e. effective area + energy dispersion + psf + background. 

One of these is an *IRF Component*.

Reply:
I just removed the "IRF" here.